flytekit/image_spec/image_spec.py

```diff
@@ -318,16 +323,30 @@ def exist(self) -> Optional[bool]:
             click.secho(f"Failed to check if the image exists with error:\n {e}", fg="red")
             return None
 
-    def _update_attribute(self, attr_name: str, values: Union[str, List[str]]) -> "ImageSpec":
+    def _update_attribute(self, attr_name: str, values: Union[str, List[str], Dict[str, Any]]) -> "ImageSpec":
```

Comment:
should the type hint here (and in the class definition) really be dict[str, Any] rather than just dict[str, str]?  What are the other types users might add?

Reply:
I can see a future, however don't have a specific example, where `int` and potentially `List[str]` be used but it seems builder dependent. Happy to constrain to `Dict[str, str]` and we could later decide to move toward `Dict[str, Any]` if we find compelling future cases.